data_handler/handlers/loan_states/zklend/events.py

```diff
@@ -303,23 +305,23 @@ def process_borrowing_event(self, event: pd.Series) -> None:
                 )
             )
 
+
     def process_repayment_event(self, event: pd.Series) -> None:
-        # The order of the values in the `data` column is: `repayer`, `beneficiary`, `token`, `raw_amount`,
-        # `face_amount`.
-        # Example: https://starkscan.co/event/0x06fa3dd6e12c9a66aeacd2eefa5a2ff2915dd1bb4207596de29bd0e8cdeeae66_5.
-        user = add_leading_zeros(event["data"][1])
-        token = add_leading_zeros(event["data"][2])
+        parsed_event = RepaymentEventSerializer.parse_event(event)
 
-        raw_amount = decimal.Decimal(str(int(event["data"][3], base=16)))
+        user = parsed_event.beneficiary
+        token = parsed_event.token
+        raw_amount = decimal.Decimal(str(int(parsed_event.raw_amount, base=16)))
```

Comment:
`decimal.Decimal(str(int(parsed_event.raw_amount, base=16)))` move this logic to `RepaymentEventSerializer`